### PR TITLE
User/bsmithvt/custom ad groups

### PR DESCRIPTION
**Pull request by bsmithvt**

### CODE REVIEW

**Comment by bsmithvt** [comment]
@michaelvidal24, good call, but we would need to enforce it only for cases where `IsDeleted=0`. Did we do this somewhere already?

**Comment by michaelvidal24** [comment]
Yes, we have an example of that in Fabric.Identity: https://github.com/HealthCatalyst/Fabric.Identity/blob/master/Fabric.Identity.SqlServer/Tables/Clients.sql.

**Comment by bsmithvt** [comment]
One thing I still need to do is handle multiple levels of group nesting. Currently only 1 level of nesting is supported.

File: Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs [hunk 1]
```diff
@@ -341,6 +356,65 @@ public async Task<Group> DeleteUserFromGroup(Group group, User user)
...
+                childToRemoveEntity.IsDeleted = true;
+            }
+
+            await AuthorizationDbContext.SaveChangesAsync();
+            await EventService.RaiseEventAsync(new EntityAuditEvent<Group>(EventTypes.ChildEntityDeletedEvent, group.Id.ToString(), group));
+
+            return group;
+        }
+
+        public Task<IEnumerable<Group>> Get(IEnumerable<string> childGroupNames)
```
Comment (michaelvidal24) [inline Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs]: Should we name this parameter something other than childGroupNames? Seems like this function is good at getting a list of groups based on a list of group names and is not really specific to childGroups...

File: Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs [hunk 2]
```diff
@@ -126,6 +140,7 @@ public async Task<IEnumerable<Group>> GetAll()
                 .ThenInclude(gr => gr.Role)
                 .Include(g => g.GroupUsers)
                 .ThenInclude(gu => gu.User)
+                .Include(g => g.ChildGroups)
                 .Where(g => !g.IsDeleted)
                 .ToArrayAsync();
```
[the remark this answers is not in the document]
Comment (bsmithvt) [reply Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs]: Good call. Will change it.

File: Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs (hunk 1 above)
Comment (bsmithvt) [reply Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs]: Good call. Will change it.

**Pull request merged**